src/js/spatial-navigation.js

```diff
@@ -0,0 +1,496 @@
+   *
+   * @param {string} key - The key representing the media action to be performed.
+   *   Accepted keys: 'play', 'pause', 'ff' (fast-forward), 'rw' (rewind).
+   */
+  performMediaAction(key) {
+    if (this.player) {
+      switch (key) {
+      case 'play':
+        if (this.player.paused()) {
+          this.player.play();
+        }
+        break;
+      case 'pause':
+        if (!this.player.paused()) {
+          this.player.pause();
+        }
+        break;
+      case 'ff':
+        this.userSeek_(this.player.currentTime() + this.STEP_SECONDS);
+        break;
+      case 'rw':
+        this.userSeek_(this.player.currentTime() - this.STEP_SECONDS);
+        break;
+      default:
+        break;
+      }
+    }
+  }
+
+  /**
+   * Prevent liveThreshold from causing seeks to seem like they
+   * are not happening from a user perspective.
+   *
+   * @param {number} ct
+   *        current time to seek to
+   */
+  userSeek_(ct) {
+    if (this.player.liveTracker && this.player.liveTracker.isLive()) {
+      this.player.liveTracker.nextSeekedFromUser();
+    }
+
+    this.player.currentTime(ct);
+  }
+
+  /**
+   * Pauses the spatial navigation functionality.
+   * This method sets a flag that can be used to temporarily disable the navigation logic.
+   */
+  pause() {
+    this.isPaused = true;
+  }
+
+  /**
+   * Resumes the spatial navigation functionality if it has been paused.
+   * This method resets the pause flag, re-enabling the navigation logic.
+   */
+  resume() {
+    this.isPaused = false;
+  }
+
+  /**
+   * Handles Player Blur.
+   *
+   */
+  handlePlayerBlur(component) {
+    if (component.name() === 'CloseButton') {
+      this.refocusComponent();
+    } else {
+      this.pause();
+
+      if (component && component.el_) {
+        this.lastFocusedComponent = component;
+      }
+    }
+  }
+
+  /**
+   * Handles Player Focus.
+   *
+   */
+  handlePlayerFocus() {
+    this.resume();
+  }
+
+  /**
+   * Gets a set of focusable components.
+   *
+   * @return {Array}
+   *         Returns an array of focusable components.
+   */
+  getComponents() {
+    const player = this.player;
+    const focusableComponents = [];
+
+    /**
+   * Searchs for children candidates.
+   *
+   * Pushes Components to array of 'focusableComponents'.
+   * Calls itself if there is children elements inside of iterated component.
+   */
+    function searchForChildrenCandidates(componentsArray) {
+      for (const i of componentsArray) {
+        if (i.hasOwnProperty('el_') && i.getIsFocusable(i.el_) && i.getIsAvailableToBeFocused(i.el_)) {
+          focusableComponents.push(i);
+        }
+        if (i.hasOwnProperty('children_') && i.children_.length > 0) {
+          searchForChildrenCandidates(i.children_);
+        }
+      }
+    }
+
+    // Iterate inside of all children components of the player.
+    player.children_.forEach((value) => {
+      if (value.hasOwnProperty('el_')) {
```

Comment:
Does the player have children that are not components?

Reply:
There can be children that are DOM elements (the video el), and children that are components but which do not have an `el_` property (MediaLoader)

Reply:
I am curious why MediaLoader is a child of the player... (if it does not represent any UI)